### PR TITLE
fix: adjust some fields sort in tables

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
fix: adjust some fields sort in tables

### Does this close any open issues?
Closes xx

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/core/models/migrationscripts/20240108_modify_fileds_sort.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		return errors.Convert(errParse)
+	}
+	db := baseRes.GetDal()
+	if u.Scheme == "mysql" {
+		// issues
+		err := db.Exec("alter table issues modify original_type varchar(500) after type;")
+		if err != nil {
+			return err
+		}
+		err = db.Exec("alter table issues modify story_point DOUBLE after original_status;")
```
Comment (Startrekzky) [inline backend/core/models/migrationscripts/20240108_modify_fileds_sort.go:45]: Move `story_point` after `priority`?
image

File: backend/core/models/migrationscripts/20240108_modify_fileds_sort.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+			return err
+		}
+		err = db.Exec("alter table pull_requests modify type varchar(100) after original_status;")
+		if err != nil {
+			return err
+		}
+		err = db.Exec("alter table pull_requests modify component varchar(100) after type;")
+		if err != nil {
+			return err
+		}
```
Comment (Startrekzky) [inline backend/core/models/migrationscripts/20240108_modify_fileds_sort.go:61]: Please also move the ref and commit_sha fields forward. The order is:

- base_repo_id
- base_ref
- base_commit_sha
- head_repo_id
- head_ref
- head_commit_sha
- merge_commit_sha

image

File: backend/core/models/migrationscripts/20240108_modify_fileds_sort.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		// cicd deployment commits
+		err = db.Exec("alter table cicd_deployment_commits modify original_status varchar(100) after status;")
+		if err != nil {
+			return err
+		}
+		err = db.Exec("alter table cicd_deployment_commits modify original_result varchar(100) after result;")
+		if err != nil {
+			return err
+		}
+		err = db.Exec("alter table cicd_deployment_commits modify duration_sec DOUBLE after finished_date;")
```
Comment (Startrekzky) [inline backend/core/models/migrationscripts/20240108_modify_fileds_sort.go:71]: The time fields' order is:
- created_date
- queued_date
- queued_duration_sec
- started_date
- finished_date
- duration_sec

image

File: backend/core/models/migrationscripts/20240108_modify_fileds_sort.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+		err = db.Exec("alter table cicd_deployment_commits modify queued_date DATETIME after duration_sec;")
+		if err != nil {
+			return err
+		}
+		err = db.Exec("alter table cicd_deployment_commits modify queued_duration_sec DOUBLE after queued_date;")
+		if err != nil {
+			return err
+		}
+		// cicd deployments
+		err = db.Exec("alter table cicd_deployments modify original_status varchar(100) after status;")
```
Comment (Startrekzky) [inline backend/core/models/migrationscripts/20240108_modify_fileds_sort.go:84]: The time fields' order is the same as the cicd_deployment_commits:

- created_date
- queued_date
- queued_duration_sec
- started_date
- finished_date
- duration_sec

**Review by Startrekzky — CHANGES_REQUESTED** [review]

**Comment by abeizn** [comment]
image

image

image

image

image

image

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**